core/role-manager/role-manager.php

```diff
@@ -155,14 +155,34 @@ private function display_role_controls( $role_slug, $role_data ) {
 	 * @access public
 	 */
 	public function get_go_pro_link_html() {
+		$promotion = $this->get_go_pro_link_content();
+
 		?>
 		<div class="elementor-role-go-pro">
-			<div class="elementor-role-go-pro__desc"><?php echo esc_html__( 'Want to give access only to content?', 'elementor' ); ?></div>
-			<div class="elementor-role-go-pro__link"><a class="elementor-button go-pro" target="_blank" href="https://go.elementor.com/go-pro-role-manager/"><?php echo esc_html__( 'Upgrade', 'elementor' ); ?></a></div>
+			<div class="elementor-role-go-pro__desc"><?php echo esc_html( $promotion['description'] ); ?></div>
+			<div class="elementor-role-go-pro__link"><a class="elementor-button go-pro" target="_blank" href="<?php esc_url( $promotion['upgrade_url'] ); ?>"><?php echo esc_html( $promotion['upgrade_text'] ); ?></a></div>
 		</div>
 		<?php
 	}
 
+	public function get_go_pro_link_content() {
+		$upgrade_url = 'https://go.elementor.com/go-pro-role-manager/';
+
+		$promotion = [
+			'description' => __( 'Want to give access only to content?' ),
+			'upgrade_url' => $upgrade_url,
+			'upgrade_text' => __( 'Upgrade' ),
+		];
+
+		$promotion = apply_filters( 'elementor/role/restrictions/custom_promotion', $promotion );
+
+		if ( strpos( $promotion['upgrade_url'], 'elementor.com' ) === false ) {
```

Comment:
can a user add elementor.com at the end of the URL ?
ex ...
https://www.google.com/?id=elementor.com

Reply:
Good point, Now only a elementor.com as the base domain works.